x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/form/field_config.tsx

```diff
@@ -361,59 +377,47 @@ export const FIELD: Record<string, FieldMeta> = {
         'How often do you want to run this test? Higher frequencies will increase your total cost.',
     }),
     dependencies: [ConfigKey.MONITOR_TYPE],
-    props: ({ dependencies }) => {
+    props: ({ dependencies }): EuiSelectProps => {
       const [monitorType] = dependencies;
       return {
         'data-test-subj': 'syntheticsMonitorConfigSchedule',
         options: monitorType === DataStream.BROWSER ? BROWSER_SCHEDULES : LIGHTWEIGHT_SCHEDULES,
+        disabled: readOnly,
       };
     },
   },
   [ConfigKey.LOCATIONS]: {
     fieldKey: ConfigKey.LOCATIONS,
     required: true,
     controlled: true,
-    component: ComboBox as React.ComponentType<EuiComboBoxProps<string>>,
+    component: ComboBox,
     label: i18n.translate('xpack.synthetics.monitorConfig.locations.label', {
       defaultMessage: 'Locations',
     }),
     helpText: i18n.translate('xpack.synthetics.monitorConfig.locations.helpText', {
       defaultMessage:
         'Where do you want to run this test from? Additional locations will increase your total cost.',
     }),
-    props: ({
-      field,
-      setValue,
-      locations,
-      formState,
-    }: {
-      field?: ControllerRenderProps;
-      setValue: UseFormReturn['setValue'];
-      locations: ServiceLocations;
-      formState: FormState<SyntheticsMonitor>;
-    }) => {
+    props: ({ field, setValue, locations, formState }) => {
       return {
         options: Object.values(locations).map((location) => ({
-          label: locations?.find((loc) => location.id === loc.id)?.label,
-          id: location.id,
-          key: location.id,
-          isServiceManaged: location.isServiceManaged,
+          label: locations?.find((loc) => location.id === loc.id)?.label || '',
+          id: location.id || '',
+          isServiceManaged: location.isServiceManaged || false,
         })),
-        selectedOptions: Object.values(field?.value as ServiceLocations).map((location) => ({
+        selectedOptions: Object.values(field?.value || {}).map((location) => ({
           color: locations.some((s) => s.id === location.id) ? 'default' : 'danger',
           label: locations?.find((loc) => location.id === loc.id)?.label ?? location.id,
-          id: location.id,
-          key: location.id,
-          isServiceManaged: location.isServiceManaged,
+          id: location.id || '',
+          isServiceManaged: location.isServiceManaged || false,
         })),
         'data-test-subj': 'syntheticsMonitorConfigLocations',
-        onChange: (updatedValues: ServiceLocations) => {
-          setValue(
-            ConfigKey.LOCATIONS,
-            updatedValues.map((location) => formatLocation(location)) as MonitorServiceLocations,
-            { shouldValidate: Boolean(formState.submitCount > 0) }
```

Comment:
why is this validation removed? this will means, it will end up storing whole location object in saved object.

Reply:
No it won't. I updated it so that the form uses it's own custom things, because the types were unhappy and I didn't want to change `formatLocation` as it's being used in other areas. You in the update that we're still only saving a subset of fields.